http-server/src/main/java/io/micronaut/http/server/exceptions/response/DefaultErrorResponseProcessor.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2017-2024 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.http.server.exceptions.response;
+
+import io.micronaut.context.annotation.Secondary;
+import io.micronaut.core.annotation.Internal;
+import io.micronaut.http.HttpMethod;
+import io.micronaut.http.MediaType;
+import io.micronaut.http.MutableHttpResponse;
+import io.micronaut.http.hateoas.JsonError;
+import jakarta.inject.Singleton;
+
+/**
+ * Default implementation of {@link ErrorResponseProcessor}.
+ * It delegates to {@link JsonBodyErrorResponseProvider} for JSON responses and to {@link HtmlBodyErrorResponseProvider} for HTML responses.
+ *
+ * @author Sergio del Amo
+ * @since 4.7.0
+ */
+@Internal
+@Singleton
+@Secondary
+final class DefaultErrorResponseProcessor implements ErrorResponseProcessor {
+    private final JsonBodyErrorResponseProvider<?> jsonBodyErrorResponseProvider;
+    private final HtmlBodyErrorResponseProvider<?> htmlBodyErrorResponseProvider;
+
+    DefaultErrorResponseProcessor(JsonBodyErrorResponseProvider<?> jsonBodyErrorResponseProvider, HtmlBodyErrorResponseProvider<?> htmlBodyErrorResponseProvider) {
+        this.jsonBodyErrorResponseProvider = jsonBodyErrorResponseProvider;
+        this.htmlBodyErrorResponseProvider = htmlBodyErrorResponseProvider;
+    }
+
+    @Override
+    public MutableHttpResponse processResponse(ErrorContext errorContext, MutableHttpResponse response) {
+        if (errorContext.getRequest().getMethod() == HttpMethod.HEAD) {
+            return (MutableHttpResponse<JsonError>) response;
+        }
+        final boolean isError = response.status().getCode() >= 400;
+        if (errorContext.getRequest().accept().stream().anyMatch(mediaType -> mediaType.equals(MediaType.TEXT_HTML_TYPE)) && isError) {
```

Comment:
flip the condition to avoid evaluating the lambda if `isError` returns false

Reply:
https://github.com/micronaut-projects/micronaut-core/pull/11210/commits/5fd2b03688f47d57440c13c4d1e8b11dbc5b48cc